monero/wallet_client.go

```diff
@@ -372,17 +546,48 @@ func (c *walletClient) Endpoint() string {
 	return c.endpoint
 }
 
+// Close kills the monero-wallet-rpc process closing the wallet. It is designed to only be
+// called a single time from a single go process.
 func (c *walletClient) Close() {
-	c.Lock()
-	defer c.Unlock()
-	if c.rpcProcess != nil {
-		p := c.rpcProcess
-		c.rpcProcess = nil
```

Comment:
We are no longer using locks, so I decided to remove that line and add a comment that `Close()` should only be called once. `c.rpcProcess` will only be nil if it was never assigned. Having said that, if you call `Close()` twice, nothing bad should happen other than a warning log that `StopWallet` failed.